IAmLegend/Assets/Scripts/GameManager.cs

```diff
@@ -52,8 +66,68 @@ void SpawnZombies()
         playerLastPosition = playerCurPosition;
     }
 
-    public void addScore(int points)
+    public void AddScore(int points)
     {
         score += points;
     }
+
+    private void EndGame()
+    {
+        HealthSystem playerHealth = player.GetComponents<HealthSystem>()[0];

```

Comment:
Hey this is an expensive operation and we do it in each iteration of the loop. GetComponent should be done once when this object is made and we can use that reference for the duration of the program